vmm/src/lib.rs

```diff
@@ -465,9 +465,17 @@ impl Vmm {
                     None => VmState::Created,
                 };
 
+                let config = Arc::clone(config);
+
+                let mut memory_actual_size = config.lock().unwrap().memory.size;
+                if let Some(vm) = &self.vm {
+                    memory_actual_size -= vm.get_balloon_actual()?;
```

Comment:
Here there's a big issue. In case you don't have a balloon, the call to vm.info is going to fail.
If you want to make sure vm.info will always provide the current memory size, when there's no balloon, you should simply return the config memory size.

Also, returning the current memory size is something that would require us to put more thoughts into it, as we should be able to report it when the user is using memory zones as well. But I guess we can keep that for some follow up PRs.

Reply:
Thanks for your remind.
I updated this part to handle the no balloon issue and zones.